hack/hook-me.sh

```diff
@@ -0,0 +1,291 @@
+      done
+      echo $(kubectl -n $namespace get svc inlets-lb -o go-template="{{ index (index  .status.loadBalancer.ingress 0).hostname }}")
+      ;;
+    *)
+      until host $(kubectl -n $namespace get svc inlets-lb -o go-template="{{ index (index  .status.loadBalancer.ingress 0).ip }}") 2>&1 > /dev/null
+      do
+        sleep 2s
+      done
+      echo $(kubectl -n $namespace get svc inlets-lb -o go-template="{{ index (index  .status.loadBalancer.ingress 0).ip }}")      ;;
+    esac
+}
+
+createServerPod(){
+namespace=${1:-}
+[[ -z $namespace ]] && echo "Please specify extension namespace!" && exit 1
+
+providerName=${2:-}
+[[ -z $providerName ]] && echo "Please specify the provider name (aws,gcp,azure,..etc.)!" && exit 1
+
+cat <<EOF | kubectl apply -f -
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    app: inlets-server
+    app.kubernetes.io/instance: provider-$providerName
+    app.kubernetes.io/name: gardener-extension-provider-$providerName
+    networking.gardener.cloud/to-dns: allowed
+    networking.gardener.cloud/to-public-networks: allowed
+  name: inlets-server
+  namespace: $namespace
+spec:
+  containers:
+  - args:
+    - "server"
+    - "-p"
+    - "8080"
+    - "-t"
+    - "21d809ed61915c9177fbceeaa87e307e766be5f2"
+    image: inlets/inlets:2.6.3
+    imagePullPolicy: IfNotPresent
+    name: inlets-server
+    resources:
+      limits:
+        cpu: 50m
+        memory: 128Mi
+      requests:
+        cpu: 20m
+        memory: 64Mi
+  - args:
+    - "server"
+    - "--target"
+    - "127.0.0.1:8080"
+    - "--listen"
+    - "0.0.0.0:9443"
+    - "--cacert"
+    - "/etc/tls/ca.crt"
+    - "--cert"
+    - "/etc/tls/tls.crt"
+    - "--key"
+    - "/etc/tls/tls.key"
+    - "--disable-authentication"
+    image:  "squareup/ghostunnel:v1.5.2"
+    imagePullPolicy: IfNotPresent
+    name: ghost-server
+    volumeMounts:
+    - name: inlets-tls
+      mountPath: "/etc/tls"
+      readOnly: true
+    resources:
+      limits:
+        cpu: 50m
+        memory: 128Mi
+      requests:
+        cpu: 20m
+        memory: 64Mi
+  - args:
+    - "sleep"
+    - "8000s"
+    image: busybox
+    imagePullPolicy: IfNotPresent
+    name: debug
+    resources:
+      limits:
+        cpu: 50m
+        memory: 128Mi
+      requests:
+        cpu: 20m
+        memory: 64Mi
+  volumes:
+  - name: inlets-tls
+    secret:
+      secretName: gardener-extension-webhook-cert
+  dnsPolicy: ClusterFirst
+  enableServiceLinks: true
+  restartPolicy: Always
+
+EOF
+}
+
+waitForInletsPodToBeReady(){
+    namespace=${1:-}
+    [[ -z $namespace ]] && echo "Please specify extension namespace!" && exit 1
+
+    until test "$(kubectl -n $namespace get pods inlets-server --no-headers | awk '{print $2}')" = "3/3"
+    do
+      sleep 2s
+    done
+}
+
+cleanUP() {
+   namespace=${1:-}
+   [[ -z $namespace ]] && echo "Please specify the extension namespace!" && exit 1
+
+   kubectl -n $namespace wait --for=delete svc/inlets-lb --timeout=60s
```

Comment:
Shouldn't we also delete the `Pod`, and if possible, revert the changes in the `Service` objects?